CMakeLists.txt

```diff
@@ -9,6 +9,22 @@ set(CMAKE_CXX_STANDARD_REQUIRED 17)
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_BUILD_TYPE Debug)
 
+
+# --- Fetch boost --------------------------------------------------------------
+option(QUICK_FTXUI_BOOST "Enable boost fetching from GitHub" ON)
+if(QUICK_FTXUI_BOOST)
+set(BOOST_ENABLE_CMAKE ON)
+include(FetchContent)
+FetchContent_Declare(
+  Boost
+  GIT_REPOSITORY https://github.com/boostorg/boost.git
```

Comment:
Can we enable only some part of boost for spirit instead of cloning all of it?

Reply:
Can we fetch only https://github.com/boostorg/spirit.git and https://github.com/boostorg/regex.git instead of fetching https://github.com/boostorg/boost.git as whole ? 
The code builds in ubuntu if we just fetch boost for spirit and regex.

Reply:
Can you try to fetch spirit only and please let me know?